web/src/app/lists/PaginatedList.js

```diff
@@ -278,15 +305,17 @@ export class PaginatedList extends React.PureComponent {
       return this.renderNoResults()
 
     const { page } = this.state
-    const { width, noPlaceholder } = this.props
+    const { itemsPerPage, width, noPlaceholder } = this.props
 
-    const items = (this.props.items || [])
-      .slice(page * ITEMS_PER_PAGE, (page + 1) * ITEMS_PER_PAGE)
-      .map(this.renderItem)
+    let items = this.props.items
+    if (!this.props.infiniteScroll) {
+      items = items.slice(page * itemsPerPage, (page + 1) * itemsPerPage)
+    }
+    items = items.map(this.renderItem)
 
     // Display full list when loading
     if (!noPlaceholder) {
-      while (this.isLoading() && items.length < ITEMS_PER_PAGE) {
+      while (this.isLoading() && items.length < itemsPerPage) {
```

Comment:
For placeholders to work the way they do otherwise, I think we need pageCount here
```suggestion
      while (this.isLoading() && items.length < itemsPerPage * this.pageCount()) {
```

Reply:
I'll look into this :+1:

Reply:
This is working as intended, this is controlling how many items to render, so we wouldn't want to render `itemsPerPage * n`, we only want to render placeholders that would match what a full list would look like for that page.

Reply:
That would mean when you get to the bottom, you would not get placeholders for the next (loading) page of results, correct?

Though that would be different from the alerts list page is today, it would be breaking from the behavior of PaginatedList (which does use placeholders when paginating) which we are migrating to.

Since it would be a simple change, let's just have it as a quick standup item and decide.

Reply:
Sure, however I don't think it's relevant for infinite scrolling lists since we are displaying a "🔄Loading..." indicator at the bottom of the list, in favor of placeholder list items (since we are appending to the current list and not rendering net-new content)